Modifying mappings with field names from basic SUDS API.

colinpmacarthur opened the pull request:
@lauraGgit — let’s discuss tomorrow!

colinpmacarthur committed:
Modifying mappings with basic SUDS API info.

lauraGgit commented on special-use/special-use-suds-mapping.json:
```diff
@@ -6,97 +6,97 @@
-          "mailingZIP": { "type": "integer", "suds-application": "Contact",  "suds-field": "Postal code" },
+          "emailAddress": { "type": "string", "basic-api-field" : "EMAIL"},
+          "mailingAddress": { "type": "string", "basic-api-field" : "ADDRESS_1"},
+          "mailingAddress2": { "type": "string", "basic-api-field" : "ADDRESS_2"},
```
it looks like `ADDRESS_3` is required, but in most cases will be blank.

colinpmacarthur replied:
Ah yes. Should we ask Xin to fix on the basic API side? It seems kind of awkward for us to say "if empty submit a space character."

lauraGgit commented on special-use/special-use-suds-mapping.json:
```diff
@@ -6,97 +6,97 @@
-      		"firstName": { "type": "string", "suds-application": "Contacts", "suds-field": "First Name"},
-      		"lastName": {	"type": "string", "suds-application": "Contact", "suds-field": "Last Name" },
+      		"firstName": { "type": "string", "basic-api-field" : "FIRST_NAME"},
+      		"lastName": {	"type": "string", "basic-api-field" : "LAST_NAME" },
```
I'm not sure how to display, but this would not be a required field if an existing `CONT_ID` is supplied.

colinpmacarthur replied:
Ah, so this is reflected in line 91 and 92. The request can either include a CONT_ID or this applicant-info block. If it does include the applicant-info block, then all these fields are required.

lauraGgit commented on special-use/special-use-suds-mapping.json:
```diff
@@ -6,97 +6,97 @@
-          "type": { "suds-application": "Contact", "suds-field": "Phone No. Type", "enum" : ["non-commercial","temp-outfitter-guide"]},
+          "areaCode": { "type": "integer", "basic-api-field": "AREA_CODE"},
+          "number": { "type": "integer", "basic-api-field": "PHONE_NUMBER"},
+          "extension": { "type": "integer", "basic-api-field": "EXTENSION"},
```
Looks like EXTENSION is an optional field.

colinpmacarthur replied:
It's a field on the FS form, so I put it in here, but it's not required. Is that OK?

lauraGgit replied:
actually this was meant to be a bit more broad, do we need to specify the optional fields?

colinpmacarthur replied:
You mean someone indicate that they're optional fields? Or not include any optional fields in the schema?

lauraGgit commented on special-use/special-use-suds-mapping.json:
```diff
@@ -6,97 +6,97 @@
-      		"firstName": { "type": "string", "suds-application": "Contacts", "suds-field": "First Name"},
-      		"lastName": {	"type": "string", "suds-application": "Contact", "suds-field": "Last Name" },
+      		"firstName": { "type": "string", "basic-api-field" : "FIRST_NAME"},
+      		"lastName": {	"type": "string", "basic-api-field" : "LAST_NAME" },
```
There is an optional `PERSON_PREFIX` field, do we want to include?

colinpmacarthur replied:
It's not on the FS form, so I don't think we can ask for it on the intake module. We can put it here just in case? /shrug

lauraGgit replied:
interesting. yes, let's put it down here.

colinpmacarthur replied:
ok, i will add!

lauraGgit commented on special-use/special-use-suds-mapping.json:
```diff
@@ -6,97 +6,97 @@
           },
           "operatingPlan": {
       			"description": "A completed Word or RTF operating plan template.",
+            "note": "See the 'not accessible to middleware' note below this one.",
```
Please change to `autopopulated by middleware not intake module." per our discussion.

lauraGgit reviewed (changes requested):
referenced post xls and dropped comments.

lauraGgit commented:
@colinpmacarthur have a few comments for you

colinpmacarthur committed:
Change "NOT_ACCESSIBLE_VIA_PUBLIC_API"

Changing “NOT_ACCESSIBLE_VIA_PUBLIC_API” to
“AUTOPOPULATED_BY_MIDDLEWARE_NOT_INTAKE_MODULE
